src/ui/pages/userroles/index.tsx

```diff
@@ -0,0 +1,189 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useContext, useEffect, useState } from "react";
+
+import { AppEnvContextProvider } from "../../contexts/AppEnvContext";
+
+import { ReactComponent as PlusIcon } from "../../../assets/plus.svg";
+import { RolesTable } from "../../components/userroles/components/RolesTable";
+
+import useRolesService from "../../../api/userroles/role";
+import { usePermissionsService } from "../../../api/userroles/role/permissions";
+import { getImageUrl } from "../../../utils";
+import Alert from "../../components/alert";
+import Button from "../../components/button";
+import CreateNewRole from "../../components/userroles/components/dialogs/CreateNewRole";
+import { Role } from "../../components/userroles/types";
+import { PopupContentContext } from "../../contexts/PopupContentContext";
+import "./index.scss";
+
+export const USERROLES_PAGINATION_LIMIT = 10;
+
+export type PaginationData = {
+	totalPages: number;
+	totalRolesCount: number;
+};
+
+export default function UserRolesList() {
+	//	boolean to check whether the roles and permissions recipe is enabled or not.
+	const [isFeatureEnabled, setIsFeatureEnabled] = useState(true);
+
+	const { getRoles } = useRolesService();
+	const { getPermissionsForRole } = usePermissionsService();
+	const { showToast } = useContext(PopupContentContext);
+
+	const [rolesRawResponse, setRolesRawResponse] = useState<string[]>([]);
+	// used to store roles with permissions data that are fetched on the client side.
+	const [roles, setRoles] = useState<Role[]>([]);
```

Comment:
type of role will change too. To set permissions array as undefined.

Reply:
done